src/reader/ns_reader.rs

```diff
@@ -526,6 +526,13 @@ impl<R: BufRead> NsReader<R> {
         // match literally the start name. See `Self::check_end_names` documentation
         self.reader.read_to_end_into(end, buf)
     }
+
+    /// Same as `read_to_end_into` but returns all the bytes read instead of the indices.
+    pub fn read_to_end_to_string(&mut self, end: QName, buf: &mut Vec<u8>) -> Result<Vec<u8>> {
+        let parser = self.parser.clone();
```

Comment:
`Parser` is a quite big structure. I would prefer to see implementation that does not copy it.